libraries/botbuilder-dialogs/tests/dialogHelper.test.js

```diff
@@ -219,4 +259,27 @@ describe('runDialog()', function() {
             await handlesBotAndSkillsTestCases(FlowTestCase.LeafSkill, true);
         });
     });
+
+    describe('HandlesEarlyDialogEndings', () => {
+        it('handles premature dialog ending', async () => {
+            const dialog = new ComponentDialogWithPrematureEnd();
+            const testFlow = createTestFlow(dialog, FlowTestCase.MiddleSkill);
+            await testFlow.send('Hi').startTest();
+
+            strictEqual(dialog.endReason, DialogReason.endCalled);
+            ok(_eocSent, 'Skills should send EndConversation to channel');
+            strictEqual(_eocSent.type, ActivityTypes.EndOfConversation);
+            strictEqual(_eocSent.code, EndOfConversationCodes.CompletedSuccessfully);
```

Comment:
These assertions are unnecessary as the scenario is reliant on a Skills scenario and this behavior is tested in the `'HandlesBotAndSkillsTestCases'` collection of tests.

The cancellation test should cover two tests and is fine as is. Thanks!

Reply:
@mdrichardson and I synced offline, we've decided to keep the test as is to ensure the linked issue is fully covered.